### PR TITLE
require python macros for building

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

We get build errors because python rpm macros are not installed by default in the build system

https://ci.suse.de/view/Manager/view/Uyuni/job/uyuni-prs-ci-tests/977/

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by mcalmer** [comment]
@agraul @meaksh This affect also packages not in our git

```
*** Looking for failed builds at Devel:Galaxy:Manager:4.2:SLE15-SUSE-Manager-Tools...
 ansible
 | mgr-custom-info
 | | mgr-osad
 | | | mgr-virtualization
 | | | | python-hwdata
 | | | | | spacecmd
 | | | | | | spacewalk-client-tools
 | | | | | | | spacewalk-oscap
 | | | | | | | | suseRegisterInfo
 | | | | | | | | | uyuni-common-libs
 | | | | | | | | | | zypp-plugin-spacewalk
 F F F F F F F F F F F  SLE_15 aarch64 (published)
 F F F F F F F F F F F  SLE_15 ppc64le (published)
 F F F F F F F F F F F  SLE_15 s390x (published)
 F F F F F F F F F F F  SLE_15 x86_64 (published)
```

**Review by meaksh — APPROVED** [review]

**Review by agraul — APPROVED** [review]
LGTM. 

Just a note, the rpm is just called `python-rpm-macros`. It provides both `python2-` and `python3-rpm-macros` symbols, we can leave it as you did it.

**Comment by mcalmer** [comment]
> LGTM.
> 
> Just a note, the rpm is just called `python-rpm-macros`. It provides both `python2-` and `python3-rpm-macros` symbols, we can leave it as you did it.

yes, I know. But I am not sure how this is on AlmaLinux. At least we have this python3-.... also running there.
And nobody knows if they rename the package when they dropped python2 completly.
In the end - however we do it, it will be wrong:-)

File: backend/spacewalk-backend.spec [hunk 1]
```diff
@@ -83,6 +83,7 @@ BuildRequires:  python3-debian
 BuildRequires:  python3-rhn-client-tools
 BuildRequires:  python3-rhnlib >= 2.5.74
 BuildRequires:  python3-rpm
+BuildRequires:  python3-rpm-macros
```
Comment (vzhestkov) [inline backend/spacewalk-backend.spec:86]: The package name is `python-rpm-macros` and it's used for all python flavors, `python3-rpm-macros` is specified as `Provides` in the package. Maybe better to use `BuildRequires: python-rpm-macros`?

Comment (mcalmer) [reply backend/spacewalk-backend.spec:86]: I keep this as we use this already somewhere and that build also on AlmaLinux. I am not sure if SUSE and AlmaLinux behave the same.

Comment (vzhestkov) [reply backend/spacewalk-backend.spec:86]: Hmm, it seems Alma really has separate `python-rpm-macros` and `python3-rpm-macros` and the packages are different.

**Pull request merged**